spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/procedures/FastForwardBranchProcedure.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.procedures;
+
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.apache.spark.sql.connector.iceberg.catalog.ProcedureParameter;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+
+public class FastForwardBranchProcedure extends BaseProcedure {
+
+  private static final ProcedureParameter[] PARAMETERS =
+      new ProcedureParameter[] {
+        ProcedureParameter.required("table", DataTypes.StringType),
+        ProcedureParameter.required("source", DataTypes.StringType),
+        ProcedureParameter.required("target", DataTypes.StringType)
+      };
+
+  private static final StructType OUTPUT_TYPE =
+      new StructType(
+          new StructField[] {
+            new StructField("updated_ref", DataTypes.LongType, false, Metadata.empty())
+          });
+
+  public static SparkProcedures.ProcedureBuilder builder() {
+    return new Builder<FastForwardBranchProcedure>() {
+      @Override
+      protected FastForwardBranchProcedure doBuild() {
+        return new FastForwardBranchProcedure(tableCatalog());
+      }
+    };
+  }
+
+  private FastForwardBranchProcedure(TableCatalog tableCatalog) {
+    super(tableCatalog);
+  }
+
+  @Override
+  public ProcedureParameter[] parameters() {
+    return PARAMETERS;
+  }
+
+  @Override
+  public StructType outputType() {
+    return OUTPUT_TYPE;
+  }
+
+  @Override
+  public InternalRow[] call(InternalRow args) {
+    Identifier tableIdent = toIdentifier(args.getString(0), PARAMETERS[0].name());
+    String source = args.getString(1);
+    String target = args.getString(2);
+
+    return modifyIcebergTable(
+        tableIdent,
+        table -> {
+          table.manageSnapshots().fastForwardBranch(source, target).commit();
```

Comment:
Is this correct? Here's the javadoc:

```java
  /**
   * Performs a fast-forward of the given target branch up to the source snapshot if target is an
   * ancestor of source. The source branch will remain unchanged, the target branch will retain its
   * retention properties.
   *
   * @param name Branch to fast-forward
   * @param source Source reference for the target to be fast forwarded to
   * @return this for method chaining
   * @throws IllegalArgumentException if the target branch is not an ancestor of source
   */
  ManageSnapshots fastForwardBranch(String name, String source);
```

Looks like it should be `fastForwardBranch(target, source)` to me.

Reply:
I do agree the naming in this method is little confusing.
As you can see in the javadoc:
` * @param name Branch to fast-forward`  -> signifies the branch that we want to fast forward, which in our case is source.

` @param source Source reference for the target to be fast forwarded to` -> Going by this param desc, my understanding is as below:
```
Source reference -> this would be the ref of the branch to which we want to fast forward. 
Target -> This is the branch that we want to fast-forward.
```

Reply:
Okay, thanks! Looks like we should rename args to that method. That is pretty confusing.